src/Strategy/Conda/CondaEnvCreate.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Conda.CondaEnvCreate (
+  analyze,
+  buildGraph,
+  CondaEnvCreateOut (..),
+  CondaEnvDep (..),
+  -- exposed for testing
+  parseCondaEnvDep,
+  parseEnvCreateDeps,
+) where
+
+import Control.Carrier.Diagnostics (Diagnostics, Has, warn)
+import Control.Monad.Combinators (some)
+import Data.Aeson (FromJSON (parseJSON), withObject, (.:))
+import Data.Either (partitionEithers)
+import Data.Foldable (traverse_)
+import Data.Map.Strict qualified as Map
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Void (Void)
+import Effect.Exec (AllowErr (Never), Command (..), Exec, execJson)
+import Graphing (Graphing, fromList)
+import Path (Abs, Dir, File, Path)
+import Text.Megaparsec (Parsec, chunk, errorBundlePretty, parse, single, takeWhile1P, try)
+import Types (
+  DepType (CondaType),
+  Dependency (..),
+  VerConstraint (CEq),
+ )
+
+-- conda env create --json --file <filename.yml> --dry-run --force
+-- runs conda and mocks the creation of an environment based on environment.yml.
+-- The command outputs json data, including the list of resolved packages.
+condaEnvCmd :: Path Abs File -> Command
+condaEnvCmd environmentYml =
+  Command
+    { cmdName = "conda"
+    , cmdArgs =
+        [ "env"
+        , "create"
+        , "--json"
+        , "--file"
+        , toText environmentYml
+        , "--dry-run"
+        , "--force"
+        ]
+    , cmdAllowErr = Never
+    }
+
+buildGraph :: [CondaEnvDep] -> Graphing Dependency
+buildGraph deps = Graphing.fromList (map toDependency deps)
+  where
+    toDependency :: CondaEnvDep -> Dependency
+    toDependency CondaEnvDep{..} =
+      Dependency
+        { dependencyType = CondaType
+        , dependencyName = "'" <> channel <> "':" <> platform <> ":" <> name
+        , dependencyVersion = Just $ CEq version
+        , dependencyLocations = []
+        , dependencyEnvironments = mempty
+        , dependencyTags = Map.empty
+        }
+
+analyze ::
+  ( Has Exec sig m
+  , Has Diagnostics sig m
+  ) =>
+  Path Abs Dir ->
+  Path Abs File ->
+  m (Graphing Dependency)
+analyze dir file = do
+  buildGraph <$> (parseEnvCreateDeps =<< execJson @CondaEnvCreateOut dir (condaEnvCmd file))
+
+parseEnvCreateDeps :: (Monad m, Has Diagnostics sig m) => CondaEnvCreateOut -> m [CondaEnvDep]
+parseEnvCreateDeps CondaEnvCreateOut{dependencies = deps} = do
+  let (lefts, rights) = partitionEithers . map (parse parseCondaEnvDep "") $ deps
```

Comment:
In the event it can't parse a package string this code will emit a warning but move on rather than failing fatally. 

Do we think this is ok? I like the idea of finding whatever information we can and doing the right thing with those, but I wonder if warnings would be visible enough in CI environments.

Reply:
I think emitting a warning instead of failing is preferable here. I think you're right that the warnings won't be readily visible in CI environments, but I'm not sure what other option we have. I think this is an issue with how visible our warnings are in general as opposed to this specific strategy.

I think it would be ideal if we could upload this as part of the CLI telemetry so that we can get alerted if this happens frequently, but I don't think that should be in scope for this PR.

Reply:
ok, sounds good. I just wasn't sure if we had made some decision about that and now I know. Thanks!